PSPs/drafts/psp-nesting.md

```diff
@@ -0,0 +1,546 @@
+4. Errors
+
+### Interfaces
+This section defines the required interface for this standard.
+
+#### Nesting::add_child(&mut self, parent_token_id: u128, child_nft: ChildNft) -> Result<(), NestingError>
+Selector: `0x1d6f5156`, first 4 bytes of `blake2b_256(Nesting::add_child)`
+```json
+{
+    "args": [
+        {
+        "label": "parent_token_id",
+        "type": {
+            "displayName": [
+            "ParentTokenId"
+            ],
+            "type": "u128"
+        }
+        },
+        {
+        "label": "child_nft",
+        "type": {
+            "displayName": [
+            "ChildNft"
+            ],
+            "type": "ChildNft"
+        }
+        }
+    ],
+    "docs": [
+        " Add a child NFT (from different collection) to the NFT in this collection.",
+        " The status of the added child is `Pending` if caller is not owner of child NFT",
+        " The status of the added child is `Accepted` if caller is is owner of child NFT",
+        " The caller needs not to be the owner of the parent_token_id,",
+        " but the Caller must be owner of the child NFT,",
+        " in order to perform transfer() ownership of the child nft to parent_token_id.",
+        "",
+        " # Requirements:",
+        " * `child_contract_address` needs to be added by collection owner",
+        " * `to_parent_token_id` must exist.",
+        " * `child_token_id` must exist.",
+        " * There cannot be two identical children.",
+        "",
+        " # Arguments:",
+        " * `to_parent_token_id`: is the tokenId of the parent NFT. The receiver of child.",
+        " * `child_nft`: (collection_id, token_id) of the child instance.",
+        "",
+        " # Result:",
+        " Ownership of child NFT will be transferred to this contract (cross contract call)",
+        " On success emits `ChildAdded`",
+        " On success emits `ChildAccepted` - only if caller is already the owner of the child NFT"
+    ],
+    "label": "Nesting::add_child",
+    "mutates": true,
+    "payable": false,
+    "returnType": {
+        "displayName": [
+        "MessageResult"
+        ],
+        "type": "Result<(), NestingError>"
+    }
+}
+```
+
+#### Nesting::remove_child(&mut self, parent_token_id: Id, child_nft: ChildNft) -> Result<(), NestingError>
+Selector: `0x27e7420e`, first 4 bytes of `blake2b_256(Nesting::remove_child)`
+```json
+{
+    "args": [
+        {
+        "label": "parent_token_id",
+        "type": {
+            "displayName": [
+            "ParentTokenId"
+            ],
+            "type": "u128"
+        }
+        },
+        {
+        "label": "child_nft",
+        "type": {
+            "displayName": [
+            "ChildNft"
+            ],
+            "type": "ChildNft"
+        }
+        }
+    ],
+    "docs": [
+        " Remove a child NFT (from different collection) from parent_token_id in this collection.",
+        " The status of added child is `Pending` if caller is not owner of child NFT",
```

Comment:
This is also not correct because child NFT owner is parent NFT

Reply:
fixed